### PR TITLE
Update Windows text rendering to use DirectWrite+Skia

### PR DESCRIPTION
**Pull request by livecodeian**
Depends on livecode-thirdparty PR: https://github.com/livecode/livecode-thirdparty/pull/99

### CODE REVIEW

File: engine/src/w32flst.cpp [hunk 1]
```diff
@@ -73,6 +73,7 @@ uint2 weighttable[] =
         FW_DONTCARE,
         FW_THIN,
         FW_EXTRALIGHT,
+		FW_EXTRALIGHT,
```
Comment (runrevmark) [inline engine/src/w32flst.cpp]: Is this meant to be duplicated? If so, please add a comment as to why.

File: libgraphics/src/directwrite-skia.cpp [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+						+ t_transform.c*t_transform.c
+						+ t_transform.d*t_transform.d) / 2);
+	
+	return S_OK;
+}
+
+template <class Callback>
+STDMETHODIMP MCGDWRenderer<Callback>::IsPixelSnappingDisabled(void* context, BOOL* isDisabled)
+{
+	// DirectWrite recommends this be enabled unless there is a good reason to disable it
```
Comment (runrevmark) [inline libgraphics/src/directwrite-skia.cpp:139]: The comment says 'recommends this be enabled', but isDisabled is being set to TRUE - could you clarify the comment?

File: libgraphics/src/directwrite-skia.cpp [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+STDMETHODIMP MCGDWRenderer<Callback>::DrawStrikethrough(void* context, FLOAT baselineOriginX, FLOAT baselineOriginY, const DWRITE_STRIKETHROUGH* strikethrough, IUnknown* clientDrawingEffect)
+{
+	// Ignored
+	return S_OK;
+}
+
+template <class Callback>
+STDMETHODIMP MCGDWRenderer<Callback>::DrawUnderline(void* context, FLOAT baselineOriginX, FLOAT baselineOriginY, const DWRITE_UNDERLINE* underline, IUnknown* clientDrawingRect)
+{
+	// Ignored
```
Comment (runrevmark) [inline libgraphics/src/directwrite-skia.cpp:169]: Is this because we do underline ourselves?

**Comment by runrevmark** [comment]
@livecodeian Could you atend to this to make sure it is mergeable and respond to the comments please? :)

File: engine/src/w32flst.cpp (hunk 1 above)
Comment (livecodeian) [reply engine/src/w32flst.cpp]: This looks like a mistake to me. @livecodefraser can you confirm?

File: libgraphics/src/directwrite-skia.cpp (hunk 2 above)
Comment (livecodeian) [reply libgraphics/src/directwrite-skia.cpp:139]: @livecodefraser can you comment please? Should this be enabled?

Comment (livecodeian) [reply libgraphics/src/directwrite-skia.cpp:139]: I have tested with this set to FALSE with no issues so this appears to be a mistake

**Comment by runrevmark** [comment]
@livecodeian: Irrespective of the comments, can we rebase this and merge in the other two related PRs, so we can merge this one? We can file any outstanding issues as bugs.

File: engine/src/w32flst.cpp (hunk 1 above)
Comment (runrevmark) [reply engine/src/w32flst.cpp]: It does seem to be - there are 10 MCFW_ constants so there should be 10 entries in this array too.

**Comment by runrevmark** [comment]
@livecode-vulcan review ok 97a7458

**Comment by livecode-vulcan** [comment]
💙 review by @runrevmark ok 97a74584e799e0b0e76b1a132907c683678a0cef

**Comment by livecode-vulcan** [comment]
😎 test success 97a74584e799e0b0e76b1a132907c683678a0cef

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/325)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/331)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/311)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/254)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/254)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/320)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/325)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/323)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/329)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/323)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/329)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/348)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/319)

**Pull request merged**